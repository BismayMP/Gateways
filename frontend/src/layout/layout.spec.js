import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Layout from './index'
jest.mock('axios')

describe('Layout Component test', () => {
  test('Snapshot testing', () => {
    const component = renderer.create(
      <Layout>
        <a href="/">welcome</a>
      </Layout>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('renders welcome link', () => {
    render(
      <Layout>
        <a href="/">welcome</a>
      </Layout>,
    )
    const linkElement = screen.getByText(/welcome/i)
    expect(linkElement).toBeInTheDocument()
  })
})
