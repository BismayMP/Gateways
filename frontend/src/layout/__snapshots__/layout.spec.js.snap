// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component test Snapshot testing 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <a
    href="/"
  >
    welcome
  </a>
</div>
`;
